openapi/pacta.yaml

```diff
@@ -1025,6 +1025,12 @@ components:
       type: string
       enum:
         - FailureCodeUNKNOWN
+    OptionalBoolean:
+      type: string
+      enum:
+        - OptionalBooleanTRUE
+        - OptionalBooleanFALSE
+        - OptionalBooleanUNSET
```

Comment:
nit: What's wrong with just a non-required boolean? IIRC, the codegen makes non-required things pointers, and `*bool` conveys the tri-state (unset, false, true) that we care about. The enum seems clunky and verbose by comparison.

Reply:
Good question, three reasons: 

Within update requests today, we treat all fields as optional - if they're unset we don't change them. If a nilable boolean value is in the actual object, there are actually four signals that could be sent about that value to an update endpoint (`do nothing`, `clear`, `set true`, `set false`). The only way around that would be to ALWAYS set these values on update, which would break our otherwise consistent paradigm of updates, and would be ripe for bugs (someone could easily call the 'updatePortfolio' method with a new name, not realizing that  they need to set these values or they will get reset.

OK, the argument then goes, why don't we represent it differently for objects and for updates? Updates might need that fourth state, but for the actual object representation, we surely only need the three states?

Well we've taken an approach to editor fields (using highly generic methods that [leverage intricate typescript features](https://github.com/RMI-PACTA/app/blob/main/frontend/lib/editor/common.ts#L17-L23) to reduce the volume of boilerplate code), and while I think that approach is a reasonable one, one downside of it is that it couples the `UpdateX` object to be a subset of the `X` object. 

While we could bend-over-backwards to make it work with this as a `*bool`, I don't think the additional value is worth its (many) costs. Happy to discuss if you disagree!

Reply:
Thanks!